src/get-data.ts

```diff
@@ -1,12 +1,21 @@
-import { getInput } from '@actions/core';
+import { getInput, setFailed } from '@actions/core';
 import { context } from '@actions/github';
 
 import getCoverage from './get-coverage';
+import throwError from './lib/error-handling';
 import { PRData } from './types';
 
-const getData = async (): Promise<PRData> => {
+const getData = async (): Promise<PRData | void> => {
   const authToken = getInput('coverageToken');
 
+  if (!authToken) {
```

Comment:
This doesn't necessarily make the action fail. We can still display the current coverage information even if we can't send it to the API. In this case we should maybe log a warning and we could also display a message at the bottom of the PR comment (only if `coverageEndpoint` is also set).